### PR TITLE
Added more player properties

### PR DESCRIPTION
**Pull request by koliva8245**
Added hero mastery tiers and the player's banner, spray, announcer, and voice line.

### CODE REVIEW

File: Heroes.ReplayParser/MPQFiles/ReplayInitData.cs [hunk 1]
```diff
@@ -277,37 +277,54 @@ public static void Parse(Replay replay, byte[] buffer)
...
-							{
-								var heroAttributeName = new string(BitConverter.GetBytes(reader.Read(32)).Select(k => (char)k).Reverse().ToArray()); // m_hero
-								reader.Read(8); // m_tier
-							}
-						}
-					}
-				}
+                    {
+                        string banner = Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_banner
+                        if (!string.IsNullOrEmpty(banner))
```
Comment (barrett777) [inline Heroes.ReplayParser/MPQFiles/ReplayInitData.cs]: Please check if userID has a value.  I may be remembering wrong, but I'm pretty sure there are replay files where userID doesn't have a value

```suggestion
                        if (!string.IsNullOrEmpty(banner) && userID.HasValue)
```

File: Heroes.ReplayParser/MPQFiles/ReplayInitData.cs [hunk 2]
```diff
@@ -277,37 +277,54 @@ public static void Parse(Replay replay, byte[] buffer)
...
+                            replay.ClientListByUserID[userID.Value].Banner = banner;
+
+                        string spray = Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_spray
+                        if (!string.IsNullOrEmpty(spray))
```
Comment (barrett777) [inline Heroes.ReplayParser/MPQFiles/ReplayInitData.cs]:
```suggestion
                        if (!string.IsNullOrEmpty(spray) && userID.HasValue)
```

File: Heroes.ReplayParser/MPQFiles/ReplayInitData.cs [hunk 3]
```diff
@@ -277,37 +277,54 @@ public static void Parse(Replay replay, byte[] buffer)
...
+                            replay.ClientListByUserID[userID.Value].Spray = spray;
+
+                        string announcer = Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_announcerPack
+                        if (!string.IsNullOrEmpty(announcer))
```
Comment (barrett777) [inline Heroes.ReplayParser/MPQFiles/ReplayInitData.cs]:
```suggestion
                        if (!string.IsNullOrEmpty(announcer) && userID.HasValue)
```

File: Heroes.ReplayParser/MPQFiles/ReplayInitData.cs [hunk 4]
```diff
@@ -277,37 +277,54 @@ public static void Parse(Replay replay, byte[] buffer)
...
+                            replay.ClientListByUserID[userID.Value].AnnouncerPack = announcer;
+
+                        string voiceLine = Encoding.UTF8.GetString(reader.ReadBlobPrecededWithLength(9)); // m_voiceLine
+                        if (!string.IsNullOrEmpty(voiceLine))
```
Comment (barrett777) [inline Heroes.ReplayParser/MPQFiles/ReplayInitData.cs]:
```suggestion
                        if (!string.IsNullOrEmpty(voiceLine) && userID.HasValue)
```

File: Heroes.ReplayParser/MPQFiles/ReplayInitData.cs [hunk 5]
```diff
@@ -277,37 +277,54 @@ public static void Parse(Replay replay, byte[] buffer)
...
+                        // m_heroMasteryTiers
+                        if (replay.ReplayBuild >= 52561)
+                        {
+                            var heroMasteryTiersLength = reader.Read(10);
+                            for (var j = 0; j < heroMasteryTiersLength; j++)
+                            {
+                                string heroAttributeName = new string(BitConverter.GetBytes(reader.Read(32)).Select(k => (char)k).Reverse().ToArray()); // m_hero
+                                int tier = (int)reader.Read(8); // m_tier
+
+                                replay.ClientListByUserID[userID.Value].HeroMasteryTiers.Add(new HeroMasteryTier()
```
Comment (barrett777) [inline Heroes.ReplayParser/MPQFiles/ReplayInitData.cs]: Also check if userID has a value before using it

**Review by barrett777 — CHANGES_REQUESTED** [review]

**Comment by koliva8245** [comment]
Done

**Review by barrett777 — APPROVED** [review]

**Pull request merged**